.github/workflows/build.yaml

```diff
@@ -34,3 +37,57 @@ jobs:
         with:
           name: install
           path: _dest/
+
+  generate-test-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.matrix.outputs.matrix }}
+    steps:
+      - id: matrix
+        uses: msys2/msys2-tests/gha-matrix-gen@main
+
+  msys2-tests:
+    needs: [build, generate-test-matrix]
+    strategy:
+      fail-fast: false
+      matrix:
+        include: ${{ fromJson(needs.generate-test-matrix.outputs.matrix) }}
+
+    name: msys2-tests ${{ matrix.msystem }}-${{ matrix.cc }}
+    runs-on: ${{ matrix.runner }}
+    env:
+      CC: ${{ matrix.cc }}
+      CXX: ${{ matrix.cxx }}
+      FC: ${{ matrix.fc }}
+    steps:
+      - id: msys2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{ matrix.msystem }}
+          update: true
+          install: ${{ matrix.packages }}
+
+      - name: Add staging repo
+        shell: msys2 {0}
+        run: |
+          sed -i '1s|^|[staging]\nServer = https://repo.msys2.org/staging/\nSigLevel = Never\n|' /etc/pacman.conf
+
+      - name: Update using staging
+        run: |
+          msys2 -c 'pacman --noconfirm -Suuy'
+          msys2 -c 'pacman --noconfirm -Suu'
```

Comment:
Ah, I just wondered why these do not tot to be in two _separate_ steps in case, say, the `msys2-runtime` is updated in the first `pacman` call: The default shell on Windows is PowerShell, which is used here.

Maybe we want an explicit `shell: pwsh` here?

Also, unless I am mistaken, an error in `pacman` invocation would not be caught unless these two lines (or something equivalent) was added:

```powershell
$ErrorActionPreference = 'Stop'
$PSNativeCommandUseErrorActionPreference = $true
```

Reply:
I copied that from msys2-tests.  I expect that it *wants* to ignore the error, because when pacman terminates things in a core update it shows up as an error.  I have used:
```yaml
      - name: Update from staging
        continue-on-error: true
        run: |
          pacman --noconfirm --noprogressbar --overwrite '*' -Syuu
      - name: Update from staging 2
        run: |
          pacman --noconfirm --noprogressbar --overwrite '*' -Syuu
```

Reply:
I guess that could work. I am always a bit fuzzy on the exact logic of `continue-on-error`: I _think_ it mistakes _any_ output to `stderr` to mean that there was an error. But that would probably also apply to my suggested change, so 🤷